### PR TITLE
Package completion for package names in directories in `[Pkg.dir(), LOAD_PATH, pwd()]`

### PR DESCRIPTION
**Pull request by dhoegh**
This is a fix for the completion of packages in LOAD_PATH mentioned in https://groups.google.com/forum/m/?fromgroups#!topic/julia-users/9lQZJlLs99M
Since completion of package names is now also works outside Pkg.dir() I have added that files should end on `".jl"`. I do not know if its appropriate to include `pwd()` in the search but it is easy to remove. cc @timholy

### CODE REVIEW

File: base/REPLCompletions.jl [hunk 1]
```diff
@@ -228,12 +228,19 @@ function completions(string, pos)
...
         s = string[startpos:pos]
         if dotpos <= startpos
-            append!(suggestions, filter(isdir(Pkg.dir()) ? readdir(Pkg.dir()) : Union(ASCIIString,UTF8String)[]) do pname
-                pname[1] != '.' &&
-                pname != "METADATA" &&
-                pname != "REQUIRE" &&
-                startswith(pname, s)
-            end)
+            for dir in [Pkg.dir(), LOAD_PATH, pwd()]
+                !isdir(dir) && continue
```
Comment (nalimilan) [inline base/REPLCompletions.jl]: Maybe more readable as `isdir(dir) || continue`?

**Comment by stevengj** [comment]
Is it possible to have a test case?

File: base/REPLCompletions.jl [hunk 2]
```diff
@@ -228,12 +228,19 @@ function completions(string, pos)
...
+                for pname in readdir(dir)
+                    if pname[1] != '.' && pname != "METADATA" &&
+                          pname != "REQUIRE" && startswith(pname, s)
```
Comment (nalimilan) [inline base/REPLCompletions.jl]: Why not align this with other lines?

Comment (dhoegh) [inline base/REPLCompletions.jl]: I'm following pep8, when a line is continued it should have extra indentation compared to what’s enclosed by the block. Then it is easily visible it is a continued line.

**Comment by dhoegh** [comment]
I have made the corrections and added test cases.

File: base/REPLCompletions.jl (hunk 2 above)
Comment (nalimilan) [inline base/REPLCompletions.jl]: Not a big deal, but it doesn't seem the rest of the code base follows this convention. We need a Julia style guide.

Comment (dhoegh) [inline base/REPLCompletions.jl]: We certainly need a Julia style guide to refer to newcomers and make the code style more consistent.

**Comment by dhoegh** [comment]
Bump are there anything preventing this from a merge?

**Pull request merged**